numba/typing/npydecl.py

```diff
@@ -126,6 +130,85 @@ def generic(self, args, kws):
         return signature(*out)
 
 
+@builtin
+class UnaryPositiveArray(AbstractTemplate):
+    '''Typing template class for +(array) expressions.  This operator is
+    special because there is no Numpy ufunc associated with it; we
+    include typing for it here (numba.typing.npydecl) because this is
+    where the remaining array operators are defined.
+    '''
+    key = "+"
+
+    def generic(self, args, kws):
+        assert not kws
+        if len(args) == 1 and isinstance(args[0], types.Array):
+            arg_ty = args[0]
+            return arg_ty.copy()(arg_ty)
+
+
+class NumpyRulesArrayOperator(Numpy_rules_ufunc):
+    _op_map = {
+         '+': "add",
+         '-': "subtract",
+         '*': "multiply",
+        '/?': "divide",
+         '/': "true_divide",
+        '//': "floor_divide",
+         '%': "remainder",
+        '**': "power",
+        '<<': "left_shift",
+        '>>': "right_shift",
+         '&': "bitwise_and",
+         '|': "bitwise_or",
+         '^': "bitwise_xor",
+    }
+
+    @property
+    def ufunc(self):
+        return getattr(numpy, self._op_map[self.key])
+
+    @classmethod
+    def install_operations(cls):
+        for op, ufunc_name in cls._op_map.items():
+            builtin(type("NumpyRulesArrayOperator_" + ufunc_name, (cls,),
+                         dict(key=op)))
+
+    def generic(self, *args, **kws):
+        '''Overloads and calls base class generic() method, returning
+        None if a TypingError occurred.
+
+        Returning None for operators is important since operators are
+        heavily overloaded, and by suppressing type errors, we allow
+        type inference to check other possibilities before giving up
+        (particularly user-defined operators).
+        '''
+        try:
+            sig = super(NumpyRulesArrayOperator, self).generic(
+                *args, **kws)
+            # Stay out of the timedelta64 range and domain; already
```

Comment:
Not as far as I know.  I only added this logic because older tests were hitting this code instead of the work already done by Graham, and it was causing unexpected behavior.